dex/farm/src/farm_token_merge.rs

```diff
@@ -71,20 +67,21 @@ pub trait FarmTokenMergeModule:
             });
         }
 
-        if replic.is_some() {
-            tokens.push(replic.unwrap());
+        if let Some(r) = replic {
+            tokens.push(r);
         }
 
         if tokens.len() == 1 {
-            return Ok(tokens.get(0).unwrap().attributes);
+            if let Some(t) = tokens.get(0) {
+                return Ok(t.attributes);
+            }
         }
 
+        let current_epoch = self.blockchain().get_block_epoch();
         let aggregated_attributes = FarmTokenAttributes {
             reward_per_share: self.aggregated_reward_per_share(&tokens),
-            entering_epoch: self.blockchain().get_block_epoch(),
-            original_entering_epoch: self.aggregated_original_entering_epoch(&tokens),
-            apr_multiplier: self.aggregated_apr_multiplier(&tokens)?,
-            with_locked_rewards: self.aggregated_with_lock_rewards(&tokens)?,
+            entering_epoch: current_epoch,
+            original_entering_epoch: current_epoch,
             initial_farming_amount: self.aggregated_initial_farming_amount(&tokens)?,
             compounded_reward: self.aggregated_compounded_reward(&tokens),
             current_farm_amount: self.aggregated_current_farm_amount(&tokens),
```

Comment:
For efficiency, consider creating a single loop over `tokens`, in which the fields of a single `FarmTokenAttributes` instance accumulate with each iteration. At the end of the loop, final steps are made to individual fields, such as `reward_per_share`.

Reply:
Consider applying the same principle on `farm_with_lock/farm_token_merge.rs`.

Reply:
Would leave merge part as it is. We have only a few elements here - not a big deal. It is safer this way.

Reply:
I will not touch this part of the code myself, as it's a pretty big change.